### PR TITLE
fix(cairo): fix meta data validation to avoid occasional error while adding new column

### PR DESCRIPTION
**Pull request by glasstiger**
Fix in table meta data validation.
Offset was not checked properly against the size of the meta data.
Offset can be equal to the mem size as after reading the last column name the offset will be pointing just after the last byte of the meta data.
Depending on the column names, this offset could be exactly the size of the allocated meta data mem.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 3 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|3|66.67%|

File: core/src/main/java/io/questdb/cairo/TableUtils.java [hunk 1]
```diff
@@ -821,16 +821,11 @@ public static void validate(
...
 
                 CharSequence name = metaMem.getStr(offset);
-
                 if (getColumnType(metaMem, i) < 0 || nameIndex.put(name, denseCount++)) {
                     offset += Vm.getStorageLength(name);
                 } else {
                     throw validationException(metaMem).put("Duplicate column: ").put(name).put(" at [").put(i).put(']');
                 }
-
-                if (offset >= memSize) {
```
Comment (ideoma) [inline core/src/main/java/io/questdb/cairo/TableUtils.java:831]: This may be useful to check that last column name does not go beyond file size

Comment (glasstiger) [reply core/src/main/java/io/questdb/cairo/TableUtils.java:831]: that is already checked at line 815.
could not construct a test where this exception is thrown so removed it.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 3 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|3|66.67%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 3 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|3|66.67%|

**Review by ideoma — APPROVED** [review]

**Pull request merged**